src/org/mozilla/javascript/NativeJavaList.java

```diff
@@ -17,6 +18,7 @@ public NativeJavaList(Scriptable scope, Object list) {
         super(scope, list, list.getClass());
         assert list instanceof List;
         this.list = (List<Object>) list;
+        setPrototype(ScriptableObject.getClassPrototype(scope, "Array"));
```

Comment:
What happens if you are using instanceof to determine if it is an array?
```js
if (value instanceof Array) {
  // some code to operate `value`
}
```
How about `Array.isArray`?

Reply:
Hello Tuchida,  `value instanceof Array` will return `true`, while `Array.isArray(value)` will return false.

I get different results in firefox

class inheritance

```javaScript
class Foo extends Array {};
var y = new Foo();
y instanceof Array; // returns true
Array.isArray(y); // returns true
```

vs. prototype

```javascript
var Bar = function(){};
Bar.prototype = Array.prototype;
var x = new Bar();
x instanceof Array; // returns true
Array.isArray(x); // returns false
```

What do you think, is ok, when we return true for `instanceof` and false for `isArray`?

Reply:
I think `javaList instanceof Array` should be `false`.

Java's `List` and JavaScript's `Array` are different, so their detailed behaviors are different.(For example, `.indexOf`)
If the existing code is written as follows, a `List` will be passed to the code that is expecting an `Array`.
```js
if (value instanceof Array) {
  // some code to operate `value`
}
```
I think it may cause compatibility issues.